src/main/java/org/opensearch/ad/ml/ThresholdingResult.java

```diff
@@ -25,13 +31,44 @@
     private final double confidence;
     private final double rcfScore;
     private long totalUpdates;
+
+    // flag indicating if the anomaly is the start of an anomaly or part of a run of
+    // anomalies
     private boolean startOfAnomaly;
+
+    // flag indicating if the time stamp is in elevated score region to be
+    // considered as anomaly
     private boolean inHighScoreRegion;
+
+    /**
+     * position of the anomaly vis a vis the current time (can be -ve) if anomaly is
+     * detected late, which can and should happen sometime; for shingle size 1; this
+     * is always 0
+     */
     private int relativeIndex;
-    private double[] currentTimeAttribution;
-    private double[] oldValues;
+
+    // a flattened version denoting the basic contribution of each input variable
+    private double[] relevantAttribution;
+
+    // oldValues is related to relativeIndex and startOfAnomaly. Read the same
+    // field comment on AnomalyResult.
+    private double[] pastValues;
+
+    // expected values, currently set to maximum 1 expected. In the future, we
+    // might give different expected values with differently likelihood. So
+    // the two-dimensional array allows us to future-proof our applications.
+    // Also, expected values correspond to oldValues if present or current input
+    // point otherwise. If oldValues is present, it will take effort to show this
+    // on UX since we found an anomaly from the past (old values).
     private double[][] expectedValuesList;
+
+    // likelihood values for the list
+    private double[] likelihoodOfValues;
```

Comment:
Is the idea there will be one likelihood value that spans a single set of expected values? In other words, for now, only one likelihood value should be expected? If so can you add some more comments here about that?

Reply:
yes, will add comments